src/keys/FileKey.cpp

```diff
@@ -225,7 +212,7 @@ void FileKey::createXMLv2(QIODevice* device, int size)
         }
         w.writeCharacters(QChar(key[i]));
     }
-    sodium_memzero(key.data(), static_cast<std::size_t>(key.capacity()));
+    Botan::secure_scrub_memory(key.data(), static_cast<std::size_t>(key.capacity()));
```

Comment:
Make `key` a `Botan::secure_vector`, then you don't need to scrub it here.

Also, we may want to disable Botan's mlock allocator for ASAN builds. See https://github.com/randombit/botan/issues/626

Reply:
I added `--disable-modules=locking_allocator` to our ASAN build flags. I think that is what you meant? Botan itself adds that flag when running sanitization of the internal library components.

Reply:
Yes.